CantoboardFramework/Keyboard/View/Candidate/CandidatePaneView.swift

```diff
@@ -379,14 +395,18 @@ class CandidatePaneView: UIControl {
     }
     
     override func layoutSubviews() {
-        guard let superview = superview else { return }
+        guard let superview = superview,
+              let layoutConstants = layoutConstants else { return }
         
         let height = mode == .row ? rowHeight : superview.bounds.height
-        let candidateViewWidth = superview.bounds.width - expandButtonWidth - directionalLayoutMargins.trailing
+        let candidateViewWidth = superview.bounds.width - expandButtonWidth - (expandButton.isHidden ? directionalLayoutMargins.trailing : 0)
+        let leftRightInset = layoutConstants.ref.candidatePaneViewLeftRightInset
         
-        let collectionViewFrame = CGRect(origin: CGPoint.zero, size: CGSize(width: candidateViewWidth, height: height))
+        let collectionViewFrame = CGRect(x: leftRightInset, y: 0, width: candidateViewWidth - leftRightInset * 2, height: height)
         if collectionView.frame != collectionViewFrame {
-            collectionView.frame = collectionViewFrame
+            UIView.animate(withDuration: collectionView.frame == .zero ? 0 : 0.25) {
```

Comment:
Could you leave this animation part in a separate PR? Also, the right part of the panel isn't expanding with the candidate pane.